cirq-core/cirq/protocols/measurement_key_protocol.py

```diff
@@ -113,20 +119,25 @@ def measurement_key(val: Any, default: Any = RaiseTypeErrorIfNotProvided):
 def _measurement_keys_from_magic_methods(val: Any) -> Optional[AbstractSet[str]]:
     """Uses the measurement key related magic methods to get the keys for this object."""
 
-    getter = getattr(val, '_measurement_keys_', None)
+    getter = getattr(val, '_measurement_keys_str_', None)
```

Comment:
Added the fallback along with a deprecation warning.

Adding the fallback to each individual magic method instead of calling the protocol. Magic methods don't have the entire context of any extra options associated with the caller protocol, calling the protocol directly from the magic method can introduce inconsistencies.